src/test/scala/nl/knaw/dans/easy/bagindex/components/DatabaseSpec.scala

```diff
@@ -119,37 +123,74 @@ class DatabaseSpec extends TestSupportFixture with BagIndexDatabaseFixture with
     database.getBagInfo(someBagId) should matchPattern { case Failure(BagIdNotFoundException(`someBagId`)) => }
   }
 
-  "getBagsWithDoi" should "return all bags with a certain DOI" in {
+  "getBagsWithIdentifier" should "return all bags with a certain DOI (identifierType 'doi')" in {
     val bagIds @ bagId1 :: bagId2 :: Nil = List.fill(2)(UUID.randomUUID())
     val times @ time1 :: time2 :: Nil = List(
       DateTime.parse("1992-07-30T16:00:00"),
       DateTime.now()
     )
     val dois @ doi1 :: _ = List.fill(2)("10.5072/dans-x6f-kf6x")
+    val urn = "urn:nbn:nl:ui:13-00-1haq"
 
     (bagIds, times, dois).zipped.toList
-      .map { case (bagId, time, doi) => database.addBagInfo(bagId, bagId, time, doi) }
+      .map { case (bagId, time, doi) => database.addBagInfo(bagId, bagId, time, doi, urn) }
+      .collectResults shouldBe a[Success[_]]
+
+    inside(database.getBagsWithIdentifier(doi1, "doi")) {
+      case Success(bags) => bags should (have size 2 and contain only(BagInfo(bagId1, bagId1, time1, doi1, urn), BagInfo(bagId2, bagId2, time2, doi1, urn)))
+    }
+  }
+
+  it should "return all bags with a certain URN (identifierType 'urn')" in {
+    val bagIds @ bagId1 :: bagId2 :: Nil = List.fill(2)(UUID.randomUUID())
+    val times @ time1 :: time2 :: Nil = List(
+      DateTime.parse("1992-07-30T16:00:00"),
+      DateTime.now()
+    )
+    val urns @ urn1 :: _ = List.fill(2)("urn:nbn:nl:ui:13-00-1haq")
```

Comment:
shouldn't the URN be unique?

Reply:
In this test there are two bagIds and both of them have the same URN